serve/mlc_serve/engine/engine_common.py

```diff
@@ -193,7 +193,7 @@ def get_requests_to_process(
                 requests.append(
                     PrefillRequest(
                         request_id=state.request_id,
-                        token_ids=state.prompt_token_ids,
+                        token_ids=state.prompt_token_ids + state.generation_sequences[0].generated_token_ids,
```

Comment:
before the parallel sampling work, we didnt maintain prompt and decode separately. So did the parallel sampling work affect the nan issue in a worse way?

The nan issue has been a problem from the beginning, so I hope it didnt.

Reply:
please add a comment here on why we add the decode tokens to a prefill step, and it needs to be updated when we add eviction support for parallel sampling

Reply:
I have not verified, but i am pretty sure that such scenario was not affected anyhow by parallel sampling because a usecase and all logic for n == 1 and reusing of prompt block remained the same

Reply:
I saw this problem even way before the parallel sampling so I suspect this has been hidden for a long time.